src/region/RegionAnnotations.tsx

```diff
@@ -88,13 +88,16 @@ export default class RegionAnnotations extends React.Component<Props, State> {
     };
 
     handleSubmit = (): void => {
-        const { createRegion, staged } = this.props;
+        const { createRegion, setActiveAnnotationId, staged } = this.props;
 
         if (!staged) {
             return;
         }
 
         createRegion(staged);
+
+        // Flush the active annotation
+        setActiveAnnotationId(null);
```

Comment:
Why is this needed?

Reply:
This keeps the old activeId being set on creation, which was causing the previously created/active annotation from being set active again, forcing the scroll back to the previous annotation.

Reply:
It seems like we have a lot of bespoke logic around how the active id is managed. How does this interact with the blur/focus events? How does it interact with the create lifecycle events (pending, fulfilled, rejected)? Should we try to centralize its handling in the reducer?